commit 9f3612f9adc290ab7c242d03c496afeed6be89d7
Author: John Layt <jlayt@kde.org>
Date:   Sat Mar 31 22:56:49 2012 +0100

    Plasma Clock: Use correct date format api
    
    Use the proper format api rather than a temp locale. KF5 prep.

diff --git a/plasma/generic/applets/digital-clock/clock.cpp b/plasma/generic/applets/digital-clock/clock.cpp
index 0af6a03..1331776 100644
--- a/plasma/generic/applets/digital-clock/clock.cpp
+++ b/plasma/generic/applets/digital-clock/clock.cpp
@@ -472,12 +472,8 @@ void Clock::paintInterface(QPainter *p, const QStyleOptionGraphicsItem *option,
         //Create the localized date string if needed
         if (m_dateStyle) {
             // JPL This needs a complete rewrite for l10n issues
-            KLocale tmpLocale(*KGlobal::locale());
-            tmpLocale.setCalendar(calendar()->calendarType()); 
-            tmpLocale.setDateFormat("%e"); // day number of the month
-            QString day = tmpLocale.formatDate(m_date);
-            tmpLocale.setDateFormat("%m"); // short form of the month
-            QString month = tmpLocale.formatDate(m_date);
+            QString day = KGlobal::locale()->calendar()->formatDate(m_date, KLocale::Day, KLocale::ShortNumber);
+            QString month = KGlobal::locale()->calendar()->formatDate(m_date, KLocale::Month, KLocale::LongNumber);
 
             if (m_dateStyle == 1) {         //compact date
                 dateString = i18nc("@label Compact date: "
